launcher/helper.cpp

```diff
@@ -145,4 +173,157 @@ void keepScreenOn(bool isEnabled)
 	iOS_utils::keepScreenOn(isEnabled);
 #endif
 }
+
+#ifdef VCMI_ANDROID
+static constexpr int  kFolderPickerReqCode = 4242;
+
+// Intent flags: READ | WRITE | PERSISTABLE | PREFIX
+static constexpr jint kIntentFlags        = 1 | 2 | 64 | 128;
+
+class FolderPickReceiver final : public QAndroidActivityResultReceiver
+{
+public:
+    std::function<void(QString)> onDone;
+
+    // One-shot result handler for ACTION_OPEN_DOCUMENT_TREE
+    void handleActivityResult(int req, int res, const QAndroidJniObject &data) override
+    {
+        auto cb = std::move(onDone); // guarantee single-use
```

Comment:
it doesn't really guarantee anything. If you call `onDone` once more from any place by mistake or not, you'll get UB.